### PR TITLE
feat: add autoware_common

### PR DESCRIPTION
**Pull request by isamu-takagi**
Add [autoware_common](https://github.com/autowarefoundation/autoware_common) repository.

### CODE REVIEW

File: .github/workflows/generate-deb-packages.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: jspricke/ros-deb-builder-action@main
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ROS_DISTRO: humble
+          DEB_DISTRO: jammy
+          SBUILD_CONF: |
+            use Dpkg::Build::Info;
+            $environment_filter = [Dpkg::Build::Info::get_build_env_allowed(), 'ROS_DISTRO'];
```
Comment (jspricke) [inline .github/workflows/generate-deb-packages.yaml:20]:
```suggestion
```

This is not needed.

File: .github/workflows/generate-deb-packages.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  generate-deb-packages:
+    runs-on: ubuntu-latest
```
Comment (jspricke) [inline .github/workflows/generate-deb-packages.yaml:11]:
```suggestion
    runs-on: ubuntu-22.04
```
This currently only works for 22.04

**Review by esteve — APPROVED** [review]

**Comment by esteve** [comment]
@isamu-takagi I applied @jspricke suggestions (thanks!) via the web interface, but unfortunately that breaks the DCO check, I'll fix that.

File: .github/workflows/generate-deb-packages.yaml (hunk 1 above)
Comment (isamu-takagi) [reply .github/workflows/generate-deb-packages.yaml]: When I tried, I got an error because `autoware_cmake` references ROS_DISTRO in CMakeLists.txt.

Comment (jspricke) [reply .github/workflows/generate-deb-packages.yaml]: That was probably before this was added: https://github.com/autowarefoundation/autoware_common/blob/main/autoware_cmake/package.xml#L15

I tested it and it works fine now.

**Comment by esteve** [comment]
@isamu-takagi I'll leave it up to you if you want to merge this PR or add more changes. Thanks!

File: .github/workflows/generate-deb-packages.yaml (hunk 1 above)
Comment (isamu-takagi) [reply .github/workflows/generate-deb-packages.yaml]: That's nice.

**Comment by isamu-takagi** [comment]
@esteve @jspricke Thank you! It worked well.
https://github.com/autowarefoundation/autoware-deb-packages/actions/runs/4043543130

**Pull request merged**